dataflow/src/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -74,6 +74,18 @@
     protected IdentityHashMap<Block, S> elseStores;
 
     /**
+     * Number of times every block has been analyzed since the last time widening was applied. Null,
+     * if maxCountBeforeWidening is -1 which implies widening isn't using for this analysis.
+     */
+    protected IdentityHashMap<Block, Integer> blockCount;
```

Comment:
Building on the comment about making maxCountBeforeWidening final: we could also make this field final, initialize it to something non-null if needed, and then reset the map instead of creating a new one on every `init`.